skimage/restoration/_denoise.py

```diff
@@ -401,19 +402,25 @@ def _denoise_tv_chambolle_nd(image, weight=0.1, eps=2.e-4, n_iter_max=200):
             out = image + d
         else:
             out = image
-        E = (d ** 2).sum()
+        E = xp.sum(d * d)
 
         # g stores the gradients of out along each axis
         # e.g. g[0] is the first order finite difference along axis 0
         slices_g = [slice(None), ] * (ndim + 1)
         for ax in range(ndim):
             slices_g[ax+1] = slice(0, -1)
             slices_g[0] = ax
-            g[tuple(slices_g)] = np.diff(out, axis=ax)
+
+            # first order difference along the axis (np.diff not in array API)
+            _at = functools.partial(utils.slice_at_axis, axis=ax)
```

Comment:
At a glance, this is similar to what [np.diff](https://github.com/numpy/numpy/blob/d64df4553e7763208e150d6039dcb6fd3d5ef0be/numpy/lib/function_base.py#L1296) does so there should not be a performance issue.

To actually make these workarounds maintainable, I see introducing a `utils.array_api` module which has `utils.array_api.diff`, `utils.array_api.astype` and all other workarounds.